extensions/send-to-readwise/src/index.tsx

```diff
@@ -0,0 +1,137 @@
+        const refs = references
+          .split(/,|，|\n/)
+          .map((ref) => ref.trim())
+          .filter(Boolean);
+        let formattedRefs = "\n**References**\n";
+        refs.forEach((ref, index) => {
+          formattedRefs += `- [[${index + 1}] ](${ref})`;
+          if (index !== refs.length - 1) {
+            formattedRefs += "\n";
+          }
+        });
+        return formattedRefs;
+      }
+      const references = formatReferences(values.reference);
+      let content = values.content;
+      if (references) {
+        content += `\n${references}`;
+      }
+
+      try {
+        const url = values.reference;
+        const note = values.note;
+        console.log("Sending request with URL:", url);
+        console.log("Sending request with Note:", note);
+        const today = new Date().toISOString().slice(0, 10);
+        const highlights: Highlight[] = [
+          {
+            title: today,
+            text: content,
+            note: note,
+          },
+        ];
+
+        const response = await fetch(READWISE_API_URL, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Token ${apiKey}`,
+          },
+          body: JSON.stringify({ highlights }),
+        });
+
+        console.log("Received response from Readwise API:", response.status, response.statusText);
+
+        if (response.ok) {
+          console.log("Sent URL:", url);
+          showToast(Toast.Style.Success, "Successfully sent to Readwise!");
+        } else {
+          const responseBody = await response.text();
+          console.error(
+            `${responseBody} -> Request failed, status code: ${response.status}, response body: ${responseBody}`
+          );
+          console.log(`API Key: ${apiKey}`);
+          console.log(`Failed URL: ${url}`);
+          showToast(Toast.Style.Failure, "Sending failed, please check the API_KEY in Raycast settings.");
+        }
+      } catch (error) {
+        console.error(error);
+        showToast(Toast.Style.Failure, "Sending failed, please check the API_KEY in Raycast settings.");
+      }
+    },
+
+    validation: {
+      content: FormValidation.Required,
+    },
+  });
+
+  return (
+    <Form
+      navigationTitle="Send to Readwise"
+      actions={
+        <ActionPanel>
+          <Action title="Sends" onAction={() => handleSubmit(values)} />
+        </ActionPanel>
+      }
+    >
+      <Form.TextArea
+        title="Highlight"
+        {...itemProps.content}
+        id="content"
+        placeholder="(Required) Write down your fleeting notes…"
+        info="🪄 Small tips! If you are reading documents on a computer, then I think using the Readwise Chrome extension is a better choice.
+⚡️ However, if you aim to record your sudden enlightenments, then using this Raycast plugin might be a better way!
+📝 Your notes will be saved under a Readwise book named after today's date. For example: 2023-09-20."
```

Comment:
Isn't it possible to add a name as well in the form or choose an existing book, article, or item? Or can one update the book on Readwise then and move an item to another book (I'm not sure that's possible)?

We could otherwise fetch a list of books and present it to the user so the user can choose themselves:

<img width="802" alt="image" src="https://github.com/raycast/extensions/assets/1043668/1771a5c3-d26a-4a8c-b590-b726fbd0166c">